pkg/autoscaler/autoscaler/autoscaler_manager.go

```diff
@@ -216,3 +233,94 @@ func (am *autoScalerManager) updateTidbClusterAutoScaler(tac *v1alpha1.TidbClust
 	}
 	return err
 }
+
+func (am *autoScalerManager) syncMonitor(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler) error {
+	if monitorRef := tc.Status.Monitor; monitorRef != nil {
+		autoTcList, err := am.getAutoScaledClusters(tac, []v1alpha1.MemberType{v1alpha1.TiDBMemberType, v1alpha1.TiKVMemberType})
+		if err != nil {
+			return err
+		}
+		clusters := sets.String{}
+		clusterMap := make(map[string]*v1alpha1.TidbCluster)
+		for _, autoTc := range autoTcList {
+			fullName := fmt.Sprintf("%s/%s", autoTc.Namespace, autoTc.Name)
+			clusters.Insert(fullName)
+			clusterMap[fullName] = autoTc
+		}
+
+		monitor, err := am.tmLister.TidbMonitors(monitorRef.Namespace).Get(monitorRef.Name)
+		if err != nil {
+			return err
+		}
+
+		monitoredClusters := sets.String{}
+		newClusterRefs := make([]v1alpha1.TidbClusterRef, 0)
+		for _, tcRef := range monitor.Spec.Clusters {
+			ns := tcRef.Namespace
+			if len(ns) == 0 {
+				ns = monitor.Namespace
+			}
+
+			fullName := fmt.Sprintf("%s/%s", ns, tcRef.Name)
+			monitoredClusters.Insert(fullName)
+			tc, err := am.tcLister.TidbClusters(ns).Get(tcRef.Name)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					continue
+				} else {
+					return err
+				}
+			}
+			tacName, ok := tc.Labels[label.AutoInstanceLabelKey]
+			if !ok || tacName != tac.Name {
+				// Not controlled by this AutoScaler, keep it
+				newClusterRefs = append(newClusterRefs, tcRef)
+				continue
+			}
+		}
+
+		updatedTm := monitor.DeepCopy()
+		toAdd := clusters.Difference(monitoredClusters)
```

Comment:
Why only add the `Difference` clusters here? You have removed all of the TCs controlled by this TAS in L277, should not we append all of the `clusters` here?

Reply:
I made a mistake. I think we should only remove those not existed.